### PR TITLE
data.azuread_groups: support the mail_enabled and security_enabled properties

### PR DESCRIPTION
**Pull request by manicminer**
Also add an acceptance helper to check values with a custom validation func

Closes: #551 

Thanks to @Threpio for initial work in #552

### CODE REVIEW

**Comment by manicminer** [comment]
Test results

Screenshot 2021-09-28 at 16 38 28

File: docs/data-sources/groups.md [hunk 1]
```diff
@@ -18,25 +18,44 @@ When authenticated with a user principal, this data source does not require any
...
+
 ## Argument Reference
 
 The following arguments are supported:
 
 * `display_names` - (Optional) The display names of the groups.
+* `mail_enabled` - (Optional) Whether the returned groups should be mail-enabled. By itself this does not exclude security-enabled groups. Setting this to `true` ensures all groups are mail-enabled, and setting to `false` ensures that all groups are _not_ mail-enabled. To ignore this filter, omit the property or set it to null. Cannot be specified together with `object_ids`.
 * `object_ids` - (Optional) The object IDs of the groups.
 * `return_all` - (Optional) A flag to denote if all groups should be fetched and returned.
+* `security_enabled` - (Optional) Whether the returned groups should be security-enabled. By itself this does not exclude security-enabled groups. Setting this to `true` ensures all groups are security-enabled, and setting to `false` ensures that all groups are _not_ security-enabled. To ignore this filter, omit the property or set it to null. Cannot be specified together with `object_ids`.
```
Comment (jackofallops) [inline docs/data-sources/groups.md:58]: copy-pasta error?
```suggestion
* `security_enabled` - (Optional) Whether the returned groups should be security-enabled. By itself this does not exclude mail-enabled groups. Setting this to `true` ensures all groups are security-enabled, and setting to `false` ensures that all groups are _not_ security-enabled. To ignore this filter, omit the property or set it to null. Cannot be specified together with `object_ids`.
```

**Review by jackofallops — APPROVED** [review]
Just one docs comment, otherwise LGTM 👍

File: docs/data-sources/groups.md (hunk 1 above)
Comment (manicminer) [reply docs/data-sources/groups.md]: Good spot, thanks!

**Pull request merged**